### PR TITLE
Remove the last small dependency on Apache Xalan

### PR DESCRIPTION
**Pull request by adamretter**
We can use the equivalent classes from Apache Xerces instead

### CODE REVIEW

**Comment by dizzzz** [comment]
In the unit tests we explicitly test for xalan 😬

**Comment by dizzzz** [comment]
interesting... java8 only 
image

**Comment by adamretter** [comment]
There is a runtime dependency on Xalan in the GeoTools libraries used by the Spatial Index. I have opened a PR to fix that here - [GEO-305](https://github.com/geotools/geotools/pull/4141). Once GEO-305 is merged, we will need to update the Spatial Index to use a newer version of the GeoTools.

**Comment by adamretter** [comment]
> it looks the PR combines 2 PRs ?

@dizzzz now fixed.

**Review by reinhapa — APPROVED** [review]

**Comment by reinhapa** [comment]
@adamretter seems there is no support for "namespace" feature: https://github.com/eXist-db/exist/actions/runs/3649983650/jobs/6165379878#step:4:13854

**Comment by adamretter** [comment]
> @adamretter seems there is no support for "namespace" feature: https://github.com/eXist-db/exist/actions/runs/3649983650/jobs/6165379878#step:4:13854

@reinhapa Yup, that is addressed in my PR https://github.com/geotools/geotools/pull/4141

**Review by dizzzz — APPROVED** [review]

**Comment by adamretter** [comment]
@dizzzz @reinhapa Okay so they released GeoTools 28.1 and I just updated this, so I am hopeful that assuming CI goes through okay, then it is now ready.

**Comment by dizzzz** [comment]
craps 
image

**Comment by adamretter** [comment]
@dizzzz No problem. I have fixed that one now. However I have hit another one in GeoTools, I will need to send them a PR as they are not handling XML Namespaces in a manner that Saxon likes:

```java
Caused by: org.xml.sax.SAXException: Parser configuration problem: namespace reporting is not enabled
	at net.sf.saxon.event.ReceivingContentHandler.getNodeName(ReceivingContentHandler.java:477)
	at net.sf.saxon.event.ReceivingContentHandler.startElement(ReceivingContentHandler.java:366)
	at org.geotools.xml.transform.QNameValidatingHandler.startElement(QNameValidatingHandler.java:61)
	at org.geotools.xml.transform.TransformerBase$ContentHandlerFilter.startElement(TransformerBase.java:367)
	at org.geotools.xml.transform.TransformerBase$TranslatorSupport._start(TransformerBase.java:822)
```

**Comment by adamretter** [comment]
Requires this second PR https://github.com/geotools/geotools/pull/4186 to be merged to GeoTools, and then a new release of GeoTools.
After that we will need to update this PR for the new release of GeoTools that includes the above GeoTools PR.

**Review by duncdrum — COMMENTED** [review]
@adamretter despite two approvals the upstream changes still need to get merged and released. And we need to get a ci run here before merging. I’ll convert this to draft in the interim. Changes look good in principal

**Comment by adamretter** [comment]
The PR https://github.com/geotools/geotools/pull/4186 to GeoTools was merged a couple of weeks ago. We are now just waiting for a new release of GeoTools that includes that fix. Then we can bump the dependency version here and we will be good to go...

File: extensions/indexes/spatial/pom.xml
Comment (dizzzz) [inline extensions/indexes/spatial/pom.xml:49]: Nice.... did not expect such a version bump works; I tried some works on it recently, and some reading....

**Review by dizzzz — CHANGES_REQUESTED** [review]
I think this is a minor issue....

image

**Comment by adamretter** [comment]
@dizzzz @reinhapa I updated it to the latest version of GeoTools (which now finally includes all the changes I sent them) - tests appear to pass - WOOOHOO :-)

**Review by dizzzz — APPROVED** [review]

**Comment by dizzzz** [comment]
nice work !

**Comment by adamretter** [comment]
Thank you so much @dizzzz - it took almost 2 years, but we finally got there! We are now Xalan free 🥳

**Comment by dizzzz** [comment]
Woooooohooo

**Pull request merged**
